src/main/java/mate/academy/dao/impl/AbstractDao.java

```diff
@@ -0,0 +1,59 @@
+package mate.academy.dao.impl;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class AbstractDao<T> {
+    protected final SessionFactory factory;
+
+    public AbstractDao(SessionFactory factory) {
+        this.factory = factory;
+    }
+
+    protected <T> T executeInsideTransaction(Function<Session, T> function) {
```

Comment:
Something wrong with this name. I prefer to use just execute or executeTransaction